### DISCUSSION TITLE
How does Single-node minio expand?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#20305] [Q&A] [answered]
### Current Information and Expected Goals:
Deploy a Minio cluster on a single k8s node with only one disk. When the Kubernetes node expand (e.g., to three nodes), I want Minio to scale up as well and have high availability capabilities. However, I want to retain the data from the original single-node deployment.

### To achieve this goal, I tested two solutions:
1. According to the documentation, if Minio is deployed in `SNSD` mode, it need to rebuild the Minio cluster when scaled up. I tried deleting the format.json or .minio.sys files on the disk to hope to preserve the previous objects. But in reality, the previous objects only existed on the disk and the new Minio cluster could not access the objects.
2. Use Minio's internal serverpool expansion. To be able to use serverpool expansion, two PVCs were deployed on one disk. The single-node Minio deployment format is: 1 server 2 drives. Then, when Kubernetes adds 2 nodes, expand the serverpool to: 3 servers 6 drives; pool-0 and pool-1 are actually on the same disk on the original node. Then decommission pool-0, wait for the migration to complete, and delete pool-0.

Also, due to limitations of the local environment, it is not possible to backup and restore the data from the single-node deployment.

I would like to know how to expand minio cluster in this situation.

**Comment by klauspost** [comment, marked as answer]
You cannot expand a SNSD as is documented.

**Reply by Clara12062** [reply]
Thank you for your reply！ yep，i know that SNSD does not allow expansion, so I want to rebuild the minio cluster. But if I don't back up the data to a third party, there is no way to keep the current objects.

So I plan to deploy the minio cluster using 1pod2PVC, which can expand the serverpool, but will sacrifice 15% of the disk capacity during the expansion period（because only one disk）. 

Both of these do not seem to meet the needs. So when expanding from a single node to three nodes, the only way is to back up the data and rebuild the minio cluster, right?
